cengage/react-magma
# fix(input): inputs with long placeholders && `isClearable` have appropriate padding

Issue: #955

## What I did

For inputs that are clearable, we now pass the input length to the input so that it only has the space for the clearable x after the input has some characters.

## Screenshots (if appropriate):
### **No Icon**
![chrome-capture-2022-10-3 (1)](https://user-images.githubusercontent.com/91160746/199816071-7cd557b8-fa8f-4345-900a-6120acb9121c.gif)
![chrome-capture-2022-10-3 (2)](https://user-images.githubusercontent.com/91160746/199816200-2ffb79ef-cb00-4b5e-8ed0-c7eb025998bb.gif)
_isClearable false_
![image](https://user-images.githubusercontent.com/91160746/199817677-5f50234c-0a78-42fa-9811-ad4667488e7c.png)


### **With Icon**
![chrome-capture-2022-10-3 (3)](https://user-images.githubusercontent.com/91160746/199816922-85f31bb0-50f3-4a71-be77-caca881c15b7.gif)
![chrome-capture-2022-10-3 (4)](https://user-images.githubusercontent.com/91160746/199817190-f774fd24-a34b-44e6-b4f4-26fc76705ed5.gif)
_isClearable false_
![image](https://user-images.githubusercontent.com/91160746/199817314-4853415d-11e7-426b-93e0-b6b2d6d9a09d.png)

### **Clickable Icon**
![image](https://user-images.githubusercontent.com/91160746/199816340-249d2471-e917-45e5-9d97-9860343e969f.png)
![image](https://user-images.githubusercontent.com/91160746/199816367-90362b06-789a-48c7-9fc2-6f63d4eb9136.png)
_isClearable false_
![image](https://user-images.githubusercontent.com/91160746/199817493-864b1d4f-3e0e-47f1-b271-b71f3d9f5b82.png)

**Search**
![chrome-capture-2022-10-3 (5)](https://user-images.githubusercontent.com/91160746/199825536-ae4cd9d2-f5ff-4470-834d-950acdf55ae0.gif)


## Checklist 
- [N/A] I have made corresponding changes to the documentation
- [N/A] I have added tests that prove my fix is effective or that my feature works

## How to test

1. Confirm inputs with a combination of props has the correct spacing.
- Try making the page smaller for smaller inputs and inserting long text.
- Try long placeholder text.
- Try with and without isClearable.
- Try the different icon positions
- etc...
2. Confirm all input types:
- Search
- Combobox
3. It looks like the following components do not support isClearable, so these can be skipped
- DatePicker
- TimePicker
- PasswordInput
